alphagov/paas-cf
# [#144515799] Make healthcheck responses more "real world"

## What

The existing response body wasn't big enough to trigger the MTU bug that was
fixed by 4dd3c3f. By increasing the size and complexity of the response body
we hope to catch more edge cases like that bug, both in our pipeline tests
which are run after a deployment and our Pingdom checks which run all of the
time.

I've used the book "Alice's Adventures in Wonderland" because:

- at 220kB (on disk) it represents a reasonable sized website
- at 68kB (on disk after gzip) it is guaranteed to still be larger than our
  MTU size of 9kB
- the content is free to use from Project Gutenberg
- it contains real HTML

I replaced `index.html` with the file `wget`ed from:

- https://www.gutenberg.org/files/11/11-h/11-h.htm

I considered using only some of the text, but it was easier to use the whole
thing, especially as it has a license at the bottom. I've left the Windows
line endings from the original file as they shouldn't affect us.

Some custom acceptance tests are included which querying the healthcheck app
as deployed to the platform by our pipeline. They check that the response
size meets the criteria described above. This should give us an early
warning of the functionality or intention of the Pingdom checks changing.

Also check response bodies in Pingdom checks.

## How to review

1. Code review
1. Deploy the branch
1. Confirm that the acceptance tests pass
1. Deploy the Pingdom checks ([documented here](https://github.com/alphagov/paas-cf/blob/master/doc/pingdom.md#setting-up-the-checks))
1. Confirm that the `PaaS HTTPS - <your_deploy_env>` check is passing
1. Modify `PaaS DB - <your_deploy_env>` to point at an environment where the healthcheck app is deployed with a DB (not dev).
1. Confirm that the `PaaS DB - <your_deploy_env>` check is passing
1. Make sure that you delete the Pingdom checks afterwards

## Who can review

Not @dcarley
